modules/util/create.py

```diff
@@ -929,7 +929,7 @@ def create_optimizer(
                 else:
                     # the group state was not saved, initialize with an empty group state
                     new_group = new_param_groups[new_group_index]
-                    new_group['params'][:len(new_group['params'])] = range(state_index, state_index + len(new_group['params']))
+                    new_group['params'][:] = range(state_index, state_index + len(new_group['params']))
```

Comment:
That's actually a really cool optimization. The `[:]` index creates a sliced view into all values of `new_group['params']` and then assigns to them from the `range()` which is a generator that produces numbers each time it's called.

The end result is that it does an in-place replace of all values without any temporary lists being created.

And the numbers it generates are equivalent to what the oldest code was doing. So it LGTM.

Reply:
Replacing the numbers in-place is cool for another reason: It matches the old code's behaviour even more pedantically.

Reply:
Here's an isolated example of the algorithm in action:

```
>>> state_index = 11
>>> new_group = {"params": ["a", "b", "c", "d", "e"]}
>>> new_group['params'][:] = range(state_index, state_index + len(new_group['params']))
>>> new_group
{'params': [11, 12, 13, 14, 15]}
```